reana_workflow_controller/rest.py

```diff
@@ -1195,24 +1215,39 @@ def set_workflow_status(workflow_id):  # noqa
         if status == START:
             return start_workflow(organization, workflow)
         else:
-            raise NotImplemented("Status {} is not supported yet"
-                                 .format(status))
+            return jsonify({"message": "Status {} is not supported yet."
+                                       .format(status)})
+    except REANAWorkflowControllerError as e:
+        return jsonify({"message": str(e)}), 409
     except KeyError as e:
         return jsonify({"message": str(e)}), 400
+    except NotImplementedError as e:
+        return jsonify({"message": str(e)}), 501
     except Exception as e:
         return jsonify({"message": str(e)}), 500
 
 
 def start_workflow(organization, workflow):
     """Start a workflow."""
-    workflow.status = WorkflowStatus.running
-    db.session.commit()
-    if workflow.type_ == 'yadage':
-        return run_yadage_workflow_from_spec(organization,
-                                             workflow)
-    elif workflow.type_ == 'cwl':
-        return run_cwl_workflow_from_spec_endpoint(organization,
-                                                   workflow)
+    if workflow.status == WorkflowStatus.created:
+        workflow.status = WorkflowStatus.running
+        db.session.commit()
+        if workflow.type_ == 'yadage':
+            return run_yadage_workflow_from_spec(organization,
+                                                 workflow)
+        elif workflow.type_ == 'cwl':
+            return run_cwl_workflow_from_spec_endpoint(organization,
+                                                       workflow)
+        else:
+            raise NotImplementedError(
+                'Workflow type {} is not supported.'.format(workflow.type_))
+    else:
+        verb = "is" if workflow.status == WorkflowStatus.running else "has"
+        message = \
```

Comment:
I think the message is fine as is (_"The response body SHOULD include enough information for the user to recognize the source of the conflict."_, [RFC2616 section 10.4.10 409 Conflict](https://tools.ietf.org/html/rfc2616#section-10.4.100)), but just want to check that is there anything more to add to the message? (_"Ideally, the response entity would include enough information for the user or user agent to fix the problem; however, that might not be possible and is not required."_, [RFC2616 section 10.4.10 409 Conflict](https://tools.ietf.org/html/rfc2616#section-10.4.100))

Does it need to say that stop the workflow or wait for it to finish or fail?
Or something like "Only workflows that are not running can be started"?

Reply:
I don't think that now we have more trustable information to give right now, once we have more available status changes we can extend it. If the rest is fine and it has been tested, it should be ready to 🚀